src/api/rolesApi.tsx

```diff
@@ -0,0 +1,75 @@
+import { spWebContext } from "api/SPWebContext";
+import { useMutation, useQuery, useQueryClient } from "@tanstack/react-query";
+import { Person } from "./UserApi";
+
+interface SPRole {
+  Id?: number;
```

Comment:
This should be required, as the Id will be on the return object from SharePoint
```suggestion
  Id: number;
```